test/apps/iodemo/ucx_wrapper.cc

```diff
@@ -118,10 +120,85 @@ void UcxContext::UcxDisconnectCallback::operator()(ucs_status_t status)
     delete this;
 }
 
-UcxContext::UcxContext(size_t iomsg_size, double connect_timeout, bool use_am) :
-    _context(NULL), _worker(NULL), _listener(NULL), _iomsg_recv_request(NULL),
-    _iomsg_buffer(iomsg_size, '\0'), _connect_timeout(connect_timeout),
-    _use_am(use_am)
+UcxContext::UcpWorkEpoll::UcpWorkEpoll(uint32_t max_events, bool edge_trigger) :
+    _epoll_fd(0),
+    _max_events(max_events),
+    _epoll_events(NULL),
+    _edge_trigger(edge_trigger)
+{
+    _epoll_fd = epoll_create1(EPOLL_CLOEXEC);
+    assert(_epoll_fd > 0);
+
+    _epoll_events = new (std::nothrow) epoll_event[max_events]();
+    assert(_epoll_events != NULL);
+}
+
+UcxContext::UcpWorkEpoll::~UcpWorkEpoll()
+{
+    delete[] _epoll_events;
+    _epoll_events = NULL;
+
+    if (_epoll_fd > 0) {
+        close(_epoll_fd);
+        _epoll_fd = 0;
+    }
+}
+
+void UcxContext::UcpWorkEpoll::add_event(int mon_fd, uint32_t event)
+{
+    ctrl_event(mon_fd, event, EPOLL_CTL_ADD);
+}
+
+void UcxContext::UcpWorkEpoll::del_event(int mon_fd, uint32_t event)
```

Comment:
Is this used?

Reply:
I'm going to use this API to removed the monitored event fd after destroying ucp_worker

Reply:
If we only has fixed event EPOLLIN, then we don't need the parameter uint32_t event.

Reply:
Maybe we should rename del_event since we are not adding event, we are adding some entry to epoll fd list.

Reply:
I see some projects use similar add_devent/del_event for the function that is use to add/del epoll fd/IN-OUT event/level-edge event.
What's the proper name need to be used in UCX?

Reply:
I'll use ```add``` & ```delete``` & ```wait``` here after talking with Bin.

Reply:
```delete``` is keyword in C++. ```delete``` -> ```del```